### ISSUE TITLE
Training doesn't stop on Colab / TPU

### ISSUE THREAD

**Issue by MathVast** [Lightning-AI/lightning#12757]
## 🐛 Bug

When training a model on Colab using TPU, I have noticed that event after completion, training doesn't stop properly and needs to be shutdown. I suspect the problem might come from the main process that keeps running while its child processes are alive and that for some reason, the child processes aren't terminated after training is done or at least, main process isn't able to receive the notification of their completion. Sometimes, the cell ends with a SIGSEGV signal, sometimes it never stops. I thought this behaviour was caused by the DataCollator as I encountered it the first time using it but even without the DataCollator the error keeps happening.

### To Reproduce

You can reproduce the error using this [Colab Notebook](https://drive.google.com/file/d/1Um_-i6wzGdwmJAFo7PW8393CtXZT2HrJ/view?usp=sharing) and the following [dataset](https://drive.google.com/file/d/1GAG8_BCZTP3gu-iC0TJctwxq7hPsU5dZ/view?usp=sharing). The dataset was created locally from the C4 dataset provided by Hugging Face and tokenized using the RobertaTokenizer for the input_ids, and the [DataCollatorForLanguageModeling](https://huggingface.co/docs/transformers/main_classes/data_collator#transformers.DataCollatorForLanguageModeling) for the masking and the labels.
Once downloaded, you have to upload the dataset to your Colab environment and specify the path in place of the variable DATA_COLAB_PATH.

### Expected behavior

After completion, training should stop by itself.

### Environment

- PyTorch Lightning Version: 1.6.0
- PyTorch Version: 1.9.1
- Python version: 3.7
- OS: Linux
- Any other relevant information:
Torch-XLA Version: 1.9
Cloud-TPU-client Version: 0.10

cc @kaushikb11 @rohitgr7

**Comment by MathVast** [comment]
After further investigations, when using Pytorch Lightning Version 1.5.10 instead of Version 1.6.0, training goes well and stops by itself.
